maidr/custom_renderers/shiny_renderer.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Optional, Callable, TypeVar, Union, List
+
+from shiny import ui
+
+import maidr
+
+T = TypeVar("T")
+
+
+def render_maidr(func: Callable[..., T]) -> Callable[..., Union[ui.Tag, ui.HTML]]:
+    """
+    Render the MAIDR plot for a given plot object obtained from Shiny.
+
+    Parameters
+    ----------
+    func: Callable[..., T]
+        The Shiny function which returns an object that can be rendered for
+        displaying the maidr plot.
+    """
+
+    def wrapper(
+        *args: Any,
+        width: Optional[str] = None,
```

Comment:
Hi @krishnanand5,
Can you automatically resize the iframe if height and width are not passed as arguments? Please feel free to refer to #64 for guidance. Considering this library's target audience, we should try to ensure the interface is as user-friendly and accessible as possible.

Apart from that the PR looks clean, good work!

As for the [comment](https://github.com/xability/py_maidr/pull/64/files#r1720523929) that you left on #64 - My observation is that Python notebooks typically have a considerable level of flexibility when it comes to handling cell rendering, which makes suitable style changes possible. Overflow characteristics, however, can be activated if pre-defined styles are used, particularly if the content is larger than the cell's dimensions.

To handle this, you may want to try utilizing Inline-JS/CSS to more accurately regulate the overflow behavior.

Reply:
- Since iframe seems to addressed here as well, I would suggest to merge this PR after #64 . 
- Also, please create a package called 'widget' under 'maidr' package (maidr/widget/shiny.py), and move the contents there. We'll be extending this further to support GUI widgets in the future.
- We are using the htmltools ui underneath, and shiny uses the same. This annotation duplicates many of the functionalities from shiny as well as from htmltools. This could be simplified by the following

```
from __future__ import annotations

from shiny.render import ui
from shiny.types import Jsonifiable

import maidr


class render_maidr(ui):
    async def render(self) -> Jsonifiable:
        value = await self.fn()
        if value is None:
            return None

        value = maidr.render(value)
        rendered = await self.transform(value)
        return rendered
```

Reply:
The requested changes have been addressed in [in this commit](https://github.com/xability/py_maidr/commit/4fdfcd119a90c7470430d0e2f158f431cfec717d).